### PR TITLE
feat: bump dependencies, add rollback guard to api

### PR DESCRIPTION
**Pull request by ozgrakkurt**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -26,9 +26,9 @@ anyhow = "1"
 
 serde_json = "1"
 
-skar-client = "0.5.2"
-skar-net-types = "0.1.1"
-skar-format = "0.2.0"
+skar-client = "0.6"
```
Comment (JasoonS) [inline Cargo.toml:29]: I'm always used to seeing the `.0` at the end, does it not make a difference?

File: Cargo.toml [hunk 2]
```diff
@@ -26,9 +26,9 @@ anyhow = "1"
...
+skar-net-types = "0.1"
```
Comment (JasoonS) [inline Cargo.toml:30]: A downgrade?

**Review by JasoonS — COMMENTED** [review]
Nice

File: Cargo.toml (hunk 1 above)
Comment (ozgrakkurt) [reply Cargo.toml:29]: It is always "^x.x.x" implicitly so if we have a minor version the patch version doesn't matter

Comment (ozgrakkurt) [reply Cargo.toml:29]: I removed them because they get confusing

File: Cargo.toml (hunk 2 above)
Comment (ozgrakkurt) [reply Cargo.toml:30]: The patch version doesn't matter if there is a minor version because it implicitly converts it into "^x.x.x"

**Pull request merged**